src/visualstates/gui/visualstates.py

```diff
@@ -306,8 +306,12 @@ def showInfo(self, title, msg):
         QMessageBox.information(self, title, msg)
 
     def generateCppAction(self):
+        self.showInfo('C++ Code Generation',
+                      'The program generates the source file in the directory of the behaviour. It will now save the behaviour.')
+        self.saveAction()
+
         stateList = []
-        if self.fileManager.hasFile():
```

Comment:
I think there is no problem with `hasFile` method. Since problem is due to the import method. `importAction` method of `visualstates.py` uses `setFullPath` method to restore the previous path. `setFullPath` checks the provided string and if the string does not have `".xml"` adds that. Therefore, even if we try to set an empty string, `".xml"` is added to the end. We should change `importAction` method. If we use `setPath` instead of `setFullPath` and keep this part same, everything should work fine.